### ISSUE TITLE
Fix OTLP/JSON to use lowerCamelCase keys

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-collector#6287]
Once the change https://github.com/open-telemetry/opentelemetry-specification/pull/2829 is released (likely in spec 1.15.0) we can update the OTLP implementations here to stop allowing original field names and only accept lowerCamelCase.

**Comment by bogdandrutu** [comment]
this is not a bug correct?

**Comment by tigrannajaryan** [comment]
I think being non-compliant with a spec is a bug :-)

**Comment by bogdandrutu** [comment]
I would not remove that for few months, since people may have already deployed apps that send the original proto names.

**Comment by tigrannajaryan** [comment]
> I would not remove that for few months, since people may have already deployed apps that send the original proto names.
> 
> Do you think that is not the case?

I agree. Let's keep for now and remove in a few months.

**Comment by mx-psi** [comment]
@tigrannajaryan open-telemetry/opentelemetry-proto/issues/476 was resolved, and this still seems to be an issue: https://github.com/open-telemetry/opentelemetry-collector/blob/df4ba7d5d0fc8554a79dc6e1938cdc64b236f10d/pdata/pmetric/json.go#L208 Did the proto issue change anything about what we want to do here?

Also, do we consider this a bug? Is this allowed after 1.0?

**Comment by mx-psi** [comment]
From the November 15th SIG meeting, there are two questions here:
1. Do we actually want to remove this?
    - This does not impose a performance penalty
    -  We don't claim pdata to be strictly compliant with the OTLP/JSON spec
    -  But the OTLP receiver must be compliant with the spec
2. If we do, is this a breaking change?
     - Changing this would not change the type signature of any Go symbol
     - It would change the behavior, it is unclear under what circumstances a change in behavior is a breaking change (e.g. we can change it for bugs)

**Comment by tigrannajaryan** [comment]
Copying the list from the Chosen Solution here:

1. We recognize this a bug that needs to be fixed.
2. We choose camelCase as canonical JSON field casing. https://github.com/open-telemetry/opentelemetry-proto/pull/468 fixed it in the spec.
3. We make sure Collector's OTLP receiver accepts both capitalizations for one year to give everyone time to transition.
4. All SDK exporters and Collector's OTLP exporter will be immediately updated to use the camelCase (if not already).

Items 3 and 4 are the one's actionable for the Collector. Number 4 needs to be acted asap (if not already) and number 3 we can do on May 9, 2024 (one year after the OTLP spec PR was merged).

**Comment by tigrannajaryan** [comment]
>Do we actually want to remove this?

I think I would prefer to remove it. It is code that needs to be maintained, it is not spec compliant and is not necessary. I feel that one year is enough grace period to deprecate it.

>If we do, is this a breaking change?

Not in the formal sense. We are simply making the implementation compliant with the spec.

**Comment by mx-psi** [comment]
> We are simply making the implementation compliant with the spec.

Okay, this is convincing to me, I filed #8974 to make it crystal clear that our `JSONMarshaler` and `JSONUnmarshaler` implementations follow the OTLP/JSON spec.

IMO this clears out the way for pdata 1.0 :)

**Comment by atoulme** [comment]
Can we close this issue at this time?

**Comment by mx-psi** [comment]
@atoulme No, we need to fix this by dropping support for lowerCamelCase on May next year.
